### PR TITLE
Lucene numHits must be > 0

### LINKED ISSUE #5983: ArgumentException: numHits must be > 0; please use TotalHitCountCollector if you just need the total hit count
image

### PR DESCRIPTION
**Pull request by infofromca**
fix #5983

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs [hunk 1]
```diff
@@ -21,15 +21,16 @@ public async Task<IList<string>> ExecuteQueryAsync(Query query, string indexName
 
             await _luceneIndexManager.SearchAsync(indexName, searcher =>
             {
-                var collector = TopScoreDocCollector.Create(end, true);
-
-                searcher.Search(query, collector);
-                var hits = collector.GetTopDocs(start, end);
-
-                foreach (var hit in hits.ScoreDocs)
+                if (end > 0)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs:24]: How do you get `end` to be `0` ?

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs:24]: When there is no index files found

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs:24]: This PR needs more work. I'm not sure this is the proper fix. It's a nice patch but that does not fix the underlying issue. If you expect to be able to search in an index there should be an index available.

**Comment by Skrypt** [comment]
This PR needs more work. I'm not sure this is the proper fix. It's a nice patch but that does not fix the underlying issue. If you expect to be able to search in an index there should be an index available.

**Comment by infofromca** [comment]
Ithinkwe can do 2 parts. 
1--merge this to make the code stronger.
2--fix the underlying part by your code.

File: src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs (hunk 1 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs:24]: Example : 

```JSON
{
  "from" : 0,
  "size": 0,
  "query": {
    "match_all": { }
  }
}
```

**Pull request closed without merging**